src/qml/patchmanager.cpp

```diff
@@ -98,6 +98,22 @@ PatchManager::PatchManager(QObject *parent)
 
     });
 
+    QDBusPendingCallWatcher *watchGetToggleServicesList = new QDBusPendingCallWatcher(m_interface->getToggleServicesList(), this);
+    connect(watchGetToggleServicesList, &QDBusPendingCallWatcher::finished, [this](QDBusPendingCallWatcher *watcher){
```

Comment:
I get no clue from the documentation that this fires any more than once. 
I think it is also true that patch updates are notified only when entering the PM settings page and the Restart Services is finicky in the sense that it doesn't show if you first open the menu. I am not 100% sure as I haven't debugged this.

For example, I had to close and re-open the Settings app to get more than one item in the RestartServicesDialog Repeater.

So I guess the call for watcher can be moved to the getter?

Reply:
Actually this call is useful in the qml page, so it could be made synchronous with no devastating consequences. 
The repeated query needs to be done for the boolean that makes the menu entry appear.

Btw, thanks, I did not know what Restart services did until today!

Reply:
> I get no clue from the documentation that this fires any more than once.
Hmm, that's suboptimal ;)

> I think it is also true that patch updates are notified only when entering the PM settings page and the Restart Services is finicky in the sense that it doesn't show if you first open the menu. I am not 100% sure as I haven't debugged this.
> 
> For example, I had to close and re-open the Settings app to get more than one item in the RestartServicesDialog Repeater.

The easiest way to get it to appear is to unapply and re-apply a patch (or preferrably, more than one from different categories).


> So I guess the call for watcher can be moved to the getter?

I'll play around with that and see how it goes. THanks for the thorough testing.

Reply:
okay, moved to getter - seems to work.

@b100dian as I didn't completely get your second comment - is that what you had in mind?

Reply:
Yes, that is exactly what I was suggesting - a plain synchronous getter. I am not sure about "seems to work" - did you also see problems with the one-off triggered callback from the original version (and existing pattern)?

(While I usually do prefer asynchronous calls, this may not be one of the cases to sweat on).

Reply:
> Yes, that is exactly what I was suggesting - a plain synchronous getter. I am not sure about "seems to work" - did you also see problems with the one-off triggered callback from the original version (and existing pattern)?

Not really. One thing I'm not sure about is Signals. Do we need one, when to fire it, when/how it is used form QML when constructing/reading the model etc.

Reply:
I think that since the model is not changed without QML calling the getter we won't need the `toggleServicesListChanged` and QML will only call the getter once when the dialog is shown up (or: every time).

So basically if the `appsNeedRestart` is working correctly for the menu to appear, the getter will too.

(I still don't have an explanation why `appsNeedRestart` works more than once though. Maybe it so happens that it is only set from false to true only once and from there the getToggleServices is called when you navigate to the page)